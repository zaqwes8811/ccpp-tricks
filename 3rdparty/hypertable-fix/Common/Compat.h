/*
 * Copyright (C) 2007-2012 Hypertable, Inc.
 *
 * This file is part of Hypertable.
 *
 * Hypertable is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or any later version.
 *
 * Hypertable is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 * 02110-1301, USA.
 */

/** @file
 * Compatibility Macros for C/C++.
 * This file contains compatibility macros for C/C++. Always include this
 * file before including any other file!
 */

#ifndef HYPERTABLE_COMPAT_H
#define HYPERTABLE_COMPAT_H

/** @addtogroup Common
 *  @{
 */

// The same stuff for C code
#include "compat-c.h"

#include <cstddef> // for std::size_t and std::ptrdiff_t

// C++ specific stuff
#ifndef BOOST_SPIRIT_THREADSAFE
#  define BOOST_SPIRIT_THREADSAFE
#endif

#define BOOST_IOSTREAMS_USE_DEPRECATED
#define BOOST_FILESYSTEM_VERSION 3
#define BOOST_FILESYSTEM_DEPRECATED     1

#include "Sweetener.h"

#define HT_UNUSED(x) static_cast<void>(x)

/** @}*/

#endif // HYPERTABLE_COMPAT_H
