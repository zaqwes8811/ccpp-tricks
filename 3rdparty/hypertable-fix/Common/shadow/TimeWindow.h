/* -*- c++ -*-
 * Copyright (C) 2007-2013 Hypertable, Inc.
 *
 * This file is part of Hypertable.
 *
 * Hypertable is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or any later version.
 *
 * Hypertable is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 * 02110-1301, USA.
 */

/// @file
/// Declarations for TimeWindow.
/// This file contains declarations for TimeWindow, a class for defining a
/// set of time windows and checking if the current time falls with any of
/// those windows.

#ifndef Common_TimeWindow_h
#define Common_TimeWindow_h

#include <Common/Crontab.h>

#include <vector>

namespace Hypertable {

  /// @addtogroup Common
  /// @{

  /// Defines a time window.
  /// This class can be used to define a time window (or set of discontiguous
  /// time windows) that can be used, for example, to define times when certain
  /// activities can take place.  It can be initialized with a set of
  /// <i>crontab</i> entries that define the time window(s) and has member
  /// functions for setting the current time and testing to see if the current
  /// time falls within the defined time window(s).  The motivating use case
  /// for this class was to allow the %Hypertable %RangeServers to define the
  /// daily period of low activity and have it change it's maintenance algorithm
  /// to be more aggressive during these periods.  Example usage:
  /// <pre>
  /// vector<String> specs;
  /// specs.push_back("* 2-5 * * *");
  /// %TimeWindow low_activity_time(specs);
  /// ...
  /// time_t now = time(0);
  /// low_activity_time.set_current_time(now);
  /// if (low_activity_time.within_window()) {
  ///   ... perform heavy mainteance activity ...
  /// }
  /// </pre>
  class TimeWindow {
  public:

    /// Default constructor.
    TimeWindow() : m_within_window(false), m_enabled(true) { }

    /// Constructor initialized with crontab specs.
    /// This function constructs a Crontab object with each of the specs in
    /// <code>crontab_specs</code> and pushes them onto #m_crontabs.  It wraps
    /// each Crontab object construction in a try/catch block and asserts with a
    /// message indicating the bad entry that caused the constructor to throw an
    /// exception.
    /// @param crontab_specs %Crontab specs that define the time window(s)
    TimeWindow(std::vector<String> crontab_specs);

    /// Logically sets the current time.
    /// This function fetches the current time (or uses <code>now</code> if
    /// non-zero) and determines if it falls within the time window by checking
    /// it against all of the Crontab objects in #m_crontabs.  For each Crontab
    /// object in #m_crontabs, it calls Crontab::next_event() and if any of them
    /// report that the next event is within 60 seconds of the current time, it
    /// sets #m_within_window to <i>true</i>, otherwise it sets it to
    /// <i>false</i>.
    /// @param now Time to use for checking if we're within the time window
    /// @return <i>true</i> if updating the current time has caused an entry
    /// or exit from the window, <i>false</i> otherwise.
    bool update_current_time(time_t now=0);

    /// Tests to see if current time is within the window.
    /// Checks to see if the current time (as set by set_current_time()) is
    /// within the time window.
    /// @return <i>true</i> if within enabled time window, <i>false</i> otherwise
    bool within_window() const { return m_within_window && m_enabled; }

    /// Enables or disables this time window.
    /// @param value <i>true</i> if enabled, <i>false</i> otherwise
    void enable_window(bool enable) { m_enabled = enable; }

    /// Indicates if the time window is enabled or disabled.
    /// @return <i>true</i> if enabled, <i>false</i> otherwise
    bool is_window_enabled() const { return m_enabled; }

  private:

    /// Vector of Crontab objects that define each subwindow
    std::vector<Crontab> m_crontabs;

    /// Set to <i>true</i> if currently within the time window
    bool m_within_window;

    /// Set to <i>true</i> if the time window is enabled
    bool m_enabled;
  };

  /// @}
}

#endif // Common_TimeWindow_h
