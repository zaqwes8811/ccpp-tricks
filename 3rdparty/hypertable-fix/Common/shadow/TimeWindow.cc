/*
 * Copyright (C) 2007-2013 Hypertable, Inc.
 *
 * This file is part of Hypertable.
 *
 * Hypertable is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or any later version.
 *
 * Hypertable is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 * 02110-1301, USA.
 */

/// @file
/// Definitions for TimeWindow.
/// This file contains definitions for TimeWindow, a class for defining a
/// set of time windows and checking if the current time falls with any of
/// those windows.

#include <Common/Compat.h>
#include "TimeWindow.h"

#include <Common/Crontab.h>
#include <Common/Error.h>
#include <Common/Logger.h>

#include <boost/algorithm/string.hpp>

#include <ctime>

using namespace Hypertable;

TimeWindow::TimeWindow(std::vector<String> crontab_specs) 
  : m_within_window(false), m_enabled(true) {

  foreach_ht (String &spec, crontab_specs) {
    try {
      boost::trim_if(spec, boost::is_any_of("\"'"));
      if (!spec.empty())
        m_crontabs.push_back( Crontab(spec) );
    }
    catch (Exception &e) {
      HT_FATALF("Problem parsing time window crontab entry (%s) - %s",
                spec.c_str(), e.what());
    }

  }
}

bool TimeWindow::update_current_time(time_t now) {
  bool window_status_on_entry = m_within_window;
  time_t next;

  if (now == 0)
    now = time(0);

  foreach_ht (Crontab &crontab, m_crontabs) {
    next = crontab.next_event(now);
    HT_ASSERT(next >= now);
    // if next event occurs within the next minute, we're within window
    if (next - now <= (time_t)60) {
      m_within_window = true;
      return window_status_on_entry != m_within_window;
    }
  }
  m_within_window = false;
  return window_status_on_entry != m_within_window;
}
