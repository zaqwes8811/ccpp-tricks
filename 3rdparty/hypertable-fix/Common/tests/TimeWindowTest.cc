/* -*- C++ -*-
 * Copyright (C) 2007-2012 Hypertable, Inc.
 *
 * This file is part of Hypertable.
 *
 * Hypertable is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or any later version.
 *
 * Hypertable is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Hypertable. If not, see <http://www.gnu.org/licenses/>
 */

#include <Common/Compat.h>
#include <Common/Error.h>
#include <Common/TimeWindow.h>

#include <boost/algorithm/string.hpp>

#include <ctime>
#include <iostream>
#include <vector>

using namespace Hypertable;
using namespace std;

int main(int argc, char *argv[]) {
  vector<String> specs;

  specs.push_back("* 2-5 * * *");
  specs.push_back("* 14-17 * * *");

  TimeWindow low_activity(specs);

  struct tm tmval;
  memset(&tmval, 0, sizeof(tmval));
  tmval.tm_mday = 25;
  tmval.tm_mon = 11;
  tmval.tm_year = 2013 - 1900;
  time_t start_time = mktime(&tmval)+1;
  

  for (time_t t=start_time; t<=start_time+86400; t+=60) {
    low_activity.update_current_time(t);
    if (low_activity.within_window())
      cout << ctime(&t);
  }

  return 0;
}
