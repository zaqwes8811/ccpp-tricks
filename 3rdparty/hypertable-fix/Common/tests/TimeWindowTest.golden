Wed Dec 25 01:59:01 2013
Wed Dec 25 02:00:01 2013
Wed Dec 25 02:01:01 2013
Wed Dec 25 02:02:01 2013
Wed Dec 25 02:03:01 2013
Wed Dec 25 02:04:01 2013
Wed Dec 25 02:05:01 2013
Wed Dec 25 02:06:01 2013
Wed Dec 25 02:07:01 2013
Wed Dec 25 02:08:01 2013
Wed Dec 25 02:09:01 2013
Wed Dec 25 02:10:01 2013
Wed Dec 25 02:11:01 2013
Wed Dec 25 02:12:01 2013
Wed Dec 25 02:13:01 2013
Wed Dec 25 02:14:01 2013
Wed Dec 25 02:15:01 2013
Wed Dec 25 02:16:01 2013
Wed Dec 25 02:17:01 2013
Wed Dec 25 02:18:01 2013
Wed Dec 25 02:19:01 2013
Wed Dec 25 02:20:01 2013
Wed Dec 25 02:21:01 2013
Wed Dec 25 02:22:01 2013
Wed Dec 25 02:23:01 2013
Wed Dec 25 02:24:01 2013
Wed Dec 25 02:25:01 2013
Wed Dec 25 02:26:01 2013
Wed Dec 25 02:27:01 2013
Wed Dec 25 02:28:01 2013
Wed Dec 25 02:29:01 2013
Wed Dec 25 02:30:01 2013
Wed Dec 25 02:31:01 2013
Wed Dec 25 02:32:01 2013
Wed Dec 25 02:33:01 2013
Wed Dec 25 02:34:01 2013
Wed Dec 25 02:35:01 2013
Wed Dec 25 02:36:01 2013
Wed Dec 25 02:37:01 2013
Wed Dec 25 02:38:01 2013
Wed Dec 25 02:39:01 2013
Wed Dec 25 02:40:01 2013
Wed Dec 25 02:41:01 2013
Wed Dec 25 02:42:01 2013
Wed Dec 25 02:43:01 2013
Wed Dec 25 02:44:01 2013
Wed Dec 25 02:45:01 2013
Wed Dec 25 02:46:01 2013
Wed Dec 25 02:47:01 2013
Wed Dec 25 02:48:01 2013
Wed Dec 25 02:49:01 2013
Wed Dec 25 02:50:01 2013
Wed Dec 25 02:51:01 2013
Wed Dec 25 02:52:01 2013
Wed Dec 25 02:53:01 2013
Wed Dec 25 02:54:01 2013
Wed Dec 25 02:55:01 2013
Wed Dec 25 02:56:01 2013
Wed Dec 25 02:57:01 2013
Wed Dec 25 02:58:01 2013
Wed Dec 25 02:59:01 2013
Wed Dec 25 03:00:01 2013
Wed Dec 25 03:01:01 2013
Wed Dec 25 03:02:01 2013
Wed Dec 25 03:03:01 2013
Wed Dec 25 03:04:01 2013
Wed Dec 25 03:05:01 2013
Wed Dec 25 03:06:01 2013
Wed Dec 25 03:07:01 2013
Wed Dec 25 03:08:01 2013
Wed Dec 25 03:09:01 2013
Wed Dec 25 03:10:01 2013
Wed Dec 25 03:11:01 2013
Wed Dec 25 03:12:01 2013
Wed Dec 25 03:13:01 2013
Wed Dec 25 03:14:01 2013
Wed Dec 25 03:15:01 2013
Wed Dec 25 03:16:01 2013
Wed Dec 25 03:17:01 2013
Wed Dec 25 03:18:01 2013
Wed Dec 25 03:19:01 2013
Wed Dec 25 03:20:01 2013
Wed Dec 25 03:21:01 2013
Wed Dec 25 03:22:01 2013
Wed Dec 25 03:23:01 2013
Wed Dec 25 03:24:01 2013
Wed Dec 25 03:25:01 2013
Wed Dec 25 03:26:01 2013
Wed Dec 25 03:27:01 2013
Wed Dec 25 03:28:01 2013
Wed Dec 25 03:29:01 2013
Wed Dec 25 03:30:01 2013
Wed Dec 25 03:31:01 2013
Wed Dec 25 03:32:01 2013
Wed Dec 25 03:33:01 2013
Wed Dec 25 03:34:01 2013
Wed Dec 25 03:35:01 2013
Wed Dec 25 03:36:01 2013
Wed Dec 25 03:37:01 2013
Wed Dec 25 03:38:01 2013
Wed Dec 25 03:39:01 2013
Wed Dec 25 03:40:01 2013
Wed Dec 25 03:41:01 2013
Wed Dec 25 03:42:01 2013
Wed Dec 25 03:43:01 2013
Wed Dec 25 03:44:01 2013
Wed Dec 25 03:45:01 2013
Wed Dec 25 03:46:01 2013
Wed Dec 25 03:47:01 2013
Wed Dec 25 03:48:01 2013
Wed Dec 25 03:49:01 2013
Wed Dec 25 03:50:01 2013
Wed Dec 25 03:51:01 2013
Wed Dec 25 03:52:01 2013
Wed Dec 25 03:53:01 2013
Wed Dec 25 03:54:01 2013
Wed Dec 25 03:55:01 2013
Wed Dec 25 03:56:01 2013
Wed Dec 25 03:57:01 2013
Wed Dec 25 03:58:01 2013
Wed Dec 25 03:59:01 2013
Wed Dec 25 04:00:01 2013
Wed Dec 25 04:01:01 2013
Wed Dec 25 04:02:01 2013
Wed Dec 25 04:03:01 2013
Wed Dec 25 04:04:01 2013
Wed Dec 25 04:05:01 2013
Wed Dec 25 04:06:01 2013
Wed Dec 25 04:07:01 2013
Wed Dec 25 04:08:01 2013
Wed Dec 25 04:09:01 2013
Wed Dec 25 04:10:01 2013
Wed Dec 25 04:11:01 2013
Wed Dec 25 04:12:01 2013
Wed Dec 25 04:13:01 2013
Wed Dec 25 04:14:01 2013
Wed Dec 25 04:15:01 2013
Wed Dec 25 04:16:01 2013
Wed Dec 25 04:17:01 2013
Wed Dec 25 04:18:01 2013
Wed Dec 25 04:19:01 2013
Wed Dec 25 04:20:01 2013
Wed Dec 25 04:21:01 2013
Wed Dec 25 04:22:01 2013
Wed Dec 25 04:23:01 2013
Wed Dec 25 04:24:01 2013
Wed Dec 25 04:25:01 2013
Wed Dec 25 04:26:01 2013
Wed Dec 25 04:27:01 2013
Wed Dec 25 04:28:01 2013
Wed Dec 25 04:29:01 2013
Wed Dec 25 04:30:01 2013
Wed Dec 25 04:31:01 2013
Wed Dec 25 04:32:01 2013
Wed Dec 25 04:33:01 2013
Wed Dec 25 04:34:01 2013
Wed Dec 25 04:35:01 2013
Wed Dec 25 04:36:01 2013
Wed Dec 25 04:37:01 2013
Wed Dec 25 04:38:01 2013
Wed Dec 25 04:39:01 2013
Wed Dec 25 04:40:01 2013
Wed Dec 25 04:41:01 2013
Wed Dec 25 04:42:01 2013
Wed Dec 25 04:43:01 2013
Wed Dec 25 04:44:01 2013
Wed Dec 25 04:45:01 2013
Wed Dec 25 04:46:01 2013
Wed Dec 25 04:47:01 2013
Wed Dec 25 04:48:01 2013
Wed Dec 25 04:49:01 2013
Wed Dec 25 04:50:01 2013
Wed Dec 25 04:51:01 2013
Wed Dec 25 04:52:01 2013
Wed Dec 25 04:53:01 2013
Wed Dec 25 04:54:01 2013
Wed Dec 25 04:55:01 2013
Wed Dec 25 04:56:01 2013
Wed Dec 25 04:57:01 2013
Wed Dec 25 04:58:01 2013
Wed Dec 25 04:59:01 2013
Wed Dec 25 05:00:01 2013
Wed Dec 25 05:01:01 2013
Wed Dec 25 05:02:01 2013
Wed Dec 25 05:03:01 2013
Wed Dec 25 05:04:01 2013
Wed Dec 25 05:05:01 2013
Wed Dec 25 05:06:01 2013
Wed Dec 25 05:07:01 2013
Wed Dec 25 05:08:01 2013
Wed Dec 25 05:09:01 2013
Wed Dec 25 05:10:01 2013
Wed Dec 25 05:11:01 2013
Wed Dec 25 05:12:01 2013
Wed Dec 25 05:13:01 2013
Wed Dec 25 05:14:01 2013
Wed Dec 25 05:15:01 2013
Wed Dec 25 05:16:01 2013
Wed Dec 25 05:17:01 2013
Wed Dec 25 05:18:01 2013
Wed Dec 25 05:19:01 2013
Wed Dec 25 05:20:01 2013
Wed Dec 25 05:21:01 2013
Wed Dec 25 05:22:01 2013
Wed Dec 25 05:23:01 2013
Wed Dec 25 05:24:01 2013
Wed Dec 25 05:25:01 2013
Wed Dec 25 05:26:01 2013
Wed Dec 25 05:27:01 2013
Wed Dec 25 05:28:01 2013
Wed Dec 25 05:29:01 2013
Wed Dec 25 05:30:01 2013
Wed Dec 25 05:31:01 2013
Wed Dec 25 05:32:01 2013
Wed Dec 25 05:33:01 2013
Wed Dec 25 05:34:01 2013
Wed Dec 25 05:35:01 2013
Wed Dec 25 05:36:01 2013
Wed Dec 25 05:37:01 2013
Wed Dec 25 05:38:01 2013
Wed Dec 25 05:39:01 2013
Wed Dec 25 05:40:01 2013
Wed Dec 25 05:41:01 2013
Wed Dec 25 05:42:01 2013
Wed Dec 25 05:43:01 2013
Wed Dec 25 05:44:01 2013
Wed Dec 25 05:45:01 2013
Wed Dec 25 05:46:01 2013
Wed Dec 25 05:47:01 2013
Wed Dec 25 05:48:01 2013
Wed Dec 25 05:49:01 2013
Wed Dec 25 05:50:01 2013
Wed Dec 25 05:51:01 2013
Wed Dec 25 05:52:01 2013
Wed Dec 25 05:53:01 2013
Wed Dec 25 05:54:01 2013
Wed Dec 25 05:55:01 2013
Wed Dec 25 05:56:01 2013
Wed Dec 25 05:57:01 2013
Wed Dec 25 05:58:01 2013
Wed Dec 25 13:59:01 2013
Wed Dec 25 14:00:01 2013
Wed Dec 25 14:01:01 2013
Wed Dec 25 14:02:01 2013
Wed Dec 25 14:03:01 2013
Wed Dec 25 14:04:01 2013
Wed Dec 25 14:05:01 2013
Wed Dec 25 14:06:01 2013
Wed Dec 25 14:07:01 2013
Wed Dec 25 14:08:01 2013
Wed Dec 25 14:09:01 2013
Wed Dec 25 14:10:01 2013
Wed Dec 25 14:11:01 2013
Wed Dec 25 14:12:01 2013
Wed Dec 25 14:13:01 2013
Wed Dec 25 14:14:01 2013
Wed Dec 25 14:15:01 2013
Wed Dec 25 14:16:01 2013
Wed Dec 25 14:17:01 2013
Wed Dec 25 14:18:01 2013
Wed Dec 25 14:19:01 2013
Wed Dec 25 14:20:01 2013
Wed Dec 25 14:21:01 2013
Wed Dec 25 14:22:01 2013
Wed Dec 25 14:23:01 2013
Wed Dec 25 14:24:01 2013
Wed Dec 25 14:25:01 2013
Wed Dec 25 14:26:01 2013
Wed Dec 25 14:27:01 2013
Wed Dec 25 14:28:01 2013
Wed Dec 25 14:29:01 2013
Wed Dec 25 14:30:01 2013
Wed Dec 25 14:31:01 2013
Wed Dec 25 14:32:01 2013
Wed Dec 25 14:33:01 2013
Wed Dec 25 14:34:01 2013
Wed Dec 25 14:35:01 2013
Wed Dec 25 14:36:01 2013
Wed Dec 25 14:37:01 2013
Wed Dec 25 14:38:01 2013
Wed Dec 25 14:39:01 2013
Wed Dec 25 14:40:01 2013
Wed Dec 25 14:41:01 2013
Wed Dec 25 14:42:01 2013
Wed Dec 25 14:43:01 2013
Wed Dec 25 14:44:01 2013
Wed Dec 25 14:45:01 2013
Wed Dec 25 14:46:01 2013
Wed Dec 25 14:47:01 2013
Wed Dec 25 14:48:01 2013
Wed Dec 25 14:49:01 2013
Wed Dec 25 14:50:01 2013
Wed Dec 25 14:51:01 2013
Wed Dec 25 14:52:01 2013
Wed Dec 25 14:53:01 2013
Wed Dec 25 14:54:01 2013
Wed Dec 25 14:55:01 2013
Wed Dec 25 14:56:01 2013
Wed Dec 25 14:57:01 2013
Wed Dec 25 14:58:01 2013
Wed Dec 25 14:59:01 2013
Wed Dec 25 15:00:01 2013
Wed Dec 25 15:01:01 2013
Wed Dec 25 15:02:01 2013
Wed Dec 25 15:03:01 2013
Wed Dec 25 15:04:01 2013
Wed Dec 25 15:05:01 2013
Wed Dec 25 15:06:01 2013
Wed Dec 25 15:07:01 2013
Wed Dec 25 15:08:01 2013
Wed Dec 25 15:09:01 2013
Wed Dec 25 15:10:01 2013
Wed Dec 25 15:11:01 2013
Wed Dec 25 15:12:01 2013
Wed Dec 25 15:13:01 2013
Wed Dec 25 15:14:01 2013
Wed Dec 25 15:15:01 2013
Wed Dec 25 15:16:01 2013
Wed Dec 25 15:17:01 2013
Wed Dec 25 15:18:01 2013
Wed Dec 25 15:19:01 2013
Wed Dec 25 15:20:01 2013
Wed Dec 25 15:21:01 2013
Wed Dec 25 15:22:01 2013
Wed Dec 25 15:23:01 2013
Wed Dec 25 15:24:01 2013
Wed Dec 25 15:25:01 2013
Wed Dec 25 15:26:01 2013
Wed Dec 25 15:27:01 2013
Wed Dec 25 15:28:01 2013
Wed Dec 25 15:29:01 2013
Wed Dec 25 15:30:01 2013
Wed Dec 25 15:31:01 2013
Wed Dec 25 15:32:01 2013
Wed Dec 25 15:33:01 2013
Wed Dec 25 15:34:01 2013
Wed Dec 25 15:35:01 2013
Wed Dec 25 15:36:01 2013
Wed Dec 25 15:37:01 2013
Wed Dec 25 15:38:01 2013
Wed Dec 25 15:39:01 2013
Wed Dec 25 15:40:01 2013
Wed Dec 25 15:41:01 2013
Wed Dec 25 15:42:01 2013
Wed Dec 25 15:43:01 2013
Wed Dec 25 15:44:01 2013
Wed Dec 25 15:45:01 2013
Wed Dec 25 15:46:01 2013
Wed Dec 25 15:47:01 2013
Wed Dec 25 15:48:01 2013
Wed Dec 25 15:49:01 2013
Wed Dec 25 15:50:01 2013
Wed Dec 25 15:51:01 2013
Wed Dec 25 15:52:01 2013
Wed Dec 25 15:53:01 2013
Wed Dec 25 15:54:01 2013
Wed Dec 25 15:55:01 2013
Wed Dec 25 15:56:01 2013
Wed Dec 25 15:57:01 2013
Wed Dec 25 15:58:01 2013
Wed Dec 25 15:59:01 2013
Wed Dec 25 16:00:01 2013
Wed Dec 25 16:01:01 2013
Wed Dec 25 16:02:01 2013
Wed Dec 25 16:03:01 2013
Wed Dec 25 16:04:01 2013
Wed Dec 25 16:05:01 2013
Wed Dec 25 16:06:01 2013
Wed Dec 25 16:07:01 2013
Wed Dec 25 16:08:01 2013
Wed Dec 25 16:09:01 2013
Wed Dec 25 16:10:01 2013
Wed Dec 25 16:11:01 2013
Wed Dec 25 16:12:01 2013
Wed Dec 25 16:13:01 2013
Wed Dec 25 16:14:01 2013
Wed Dec 25 16:15:01 2013
Wed Dec 25 16:16:01 2013
Wed Dec 25 16:17:01 2013
Wed Dec 25 16:18:01 2013
Wed Dec 25 16:19:01 2013
Wed Dec 25 16:20:01 2013
Wed Dec 25 16:21:01 2013
Wed Dec 25 16:22:01 2013
Wed Dec 25 16:23:01 2013
Wed Dec 25 16:24:01 2013
Wed Dec 25 16:25:01 2013
Wed Dec 25 16:26:01 2013
Wed Dec 25 16:27:01 2013
Wed Dec 25 16:28:01 2013
Wed Dec 25 16:29:01 2013
Wed Dec 25 16:30:01 2013
Wed Dec 25 16:31:01 2013
Wed Dec 25 16:32:01 2013
Wed Dec 25 16:33:01 2013
Wed Dec 25 16:34:01 2013
Wed Dec 25 16:35:01 2013
Wed Dec 25 16:36:01 2013
Wed Dec 25 16:37:01 2013
Wed Dec 25 16:38:01 2013
Wed Dec 25 16:39:01 2013
Wed Dec 25 16:40:01 2013
Wed Dec 25 16:41:01 2013
Wed Dec 25 16:42:01 2013
Wed Dec 25 16:43:01 2013
Wed Dec 25 16:44:01 2013
Wed Dec 25 16:45:01 2013
Wed Dec 25 16:46:01 2013
Wed Dec 25 16:47:01 2013
Wed Dec 25 16:48:01 2013
Wed Dec 25 16:49:01 2013
Wed Dec 25 16:50:01 2013
Wed Dec 25 16:51:01 2013
Wed Dec 25 16:52:01 2013
Wed Dec 25 16:53:01 2013
Wed Dec 25 16:54:01 2013
Wed Dec 25 16:55:01 2013
Wed Dec 25 16:56:01 2013
Wed Dec 25 16:57:01 2013
Wed Dec 25 16:58:01 2013
Wed Dec 25 16:59:01 2013
Wed Dec 25 17:00:01 2013
Wed Dec 25 17:01:01 2013
Wed Dec 25 17:02:01 2013
Wed Dec 25 17:03:01 2013
Wed Dec 25 17:04:01 2013
Wed Dec 25 17:05:01 2013
Wed Dec 25 17:06:01 2013
Wed Dec 25 17:07:01 2013
Wed Dec 25 17:08:01 2013
Wed Dec 25 17:09:01 2013
Wed Dec 25 17:10:01 2013
Wed Dec 25 17:11:01 2013
Wed Dec 25 17:12:01 2013
Wed Dec 25 17:13:01 2013
Wed Dec 25 17:14:01 2013
Wed Dec 25 17:15:01 2013
Wed Dec 25 17:16:01 2013
Wed Dec 25 17:17:01 2013
Wed Dec 25 17:18:01 2013
Wed Dec 25 17:19:01 2013
Wed Dec 25 17:20:01 2013
Wed Dec 25 17:21:01 2013
Wed Dec 25 17:22:01 2013
Wed Dec 25 17:23:01 2013
Wed Dec 25 17:24:01 2013
Wed Dec 25 17:25:01 2013
Wed Dec 25 17:26:01 2013
Wed Dec 25 17:27:01 2013
Wed Dec 25 17:28:01 2013
Wed Dec 25 17:29:01 2013
Wed Dec 25 17:30:01 2013
Wed Dec 25 17:31:01 2013
Wed Dec 25 17:32:01 2013
Wed Dec 25 17:33:01 2013
Wed Dec 25 17:34:01 2013
Wed Dec 25 17:35:01 2013
Wed Dec 25 17:36:01 2013
Wed Dec 25 17:37:01 2013
Wed Dec 25 17:38:01 2013
Wed Dec 25 17:39:01 2013
Wed Dec 25 17:40:01 2013
Wed Dec 25 17:41:01 2013
Wed Dec 25 17:42:01 2013
Wed Dec 25 17:43:01 2013
Wed Dec 25 17:44:01 2013
Wed Dec 25 17:45:01 2013
Wed Dec 25 17:46:01 2013
Wed Dec 25 17:47:01 2013
Wed Dec 25 17:48:01 2013
Wed Dec 25 17:49:01 2013
Wed Dec 25 17:50:01 2013
Wed Dec 25 17:51:01 2013
Wed Dec 25 17:52:01 2013
Wed Dec 25 17:53:01 2013
Wed Dec 25 17:54:01 2013
Wed Dec 25 17:55:01 2013
Wed Dec 25 17:56:01 2013
Wed Dec 25 17:57:01 2013
Wed Dec 25 17:58:01 2013
